src/workerd/io/worker.c++

```diff
@@ -158,6 +160,20 @@ const capnp::JsonCodec& getCdpJsonCodec() {
   static const kj::Own<capnp::JsonCodec> codec = makeCdpJsonCodec();
   return *codec;
 }
+
+// Returns whether we can output color to the console. Even if this returns `true`,
+// we'll only write color codes if the output file is a TTY.
+// TODO(someday): adopt more of Node.js's checks:
+//  https://github.com/nodejs/node/blob/ac2a68c/lib/internal/tty.js#L106
+static bool permitsColor() {
```

Comment:
non-blocking nit: workerd.c++ is already fairly big. Let's move this out to a utility header file in `workerd/utils`